mne/utils/config.py

```diff
@@ -263,8 +263,9 @@ def set_config(key, value, home_dir=None, set_env=True):
     _validate_type(key, 'str', "key")
     # While JSON allow non-string types, we allow users to override config
     # settings using env, which are strings, so we enforce that here
-    _validate_type(value, (str, type(None)), "value",
-                   "None or string")
+    _validate_type(value, (str, 'path-like', type(None)), 'value')
```

Comment:
you actually get a better error message if you pass `None` directly here:
```
>>> mne.utils._validate_type(1, None, 'test')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/larsoner/python/mne-python/mne/utils/check.py", line 380, in _validate_type
    raise TypeError('%s must be an instance of %s, got %s instead'
TypeError: test must be an instance of None, got <class 'int'> instead
>>> mne.utils._validate_type(1, type(None), 'test')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/larsoner/python/mne-python/mne/utils/check.py", line 380, in _validate_type
    raise TypeError('%s must be an instance of %s, got %s instead'
TypeError: test must be an instance of NoneType, got <class 'int'> instead
```
but we can always tweak this later